src/python_testing/TC_AVSM_2_7.py

```diff
@@ -0,0 +1,339 @@
+        )
+        logger.info(f"Rx'd RateDistortionTradeOffPoints: {aRateDistortionTradeOffPoints}")
+
+        self.step(5)
+        aMinViewport = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MinViewport
+        )
+        logger.info(f"Rx'd MinViewport: {aMinViewport}")
+
+        self.step(6)
+        aVideoSensorParams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.VideoSensorParams
+        )
+        logger.info(f"Rx'd VideoSensorParams: {aVideoSensorParams}")
+
+        self.step(7)
+        aMaxEncodedPixelRate = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MaxEncodedPixelRate
+        )
+        logger.info(f"Rx'd MaxEncodedPixelRate: {aMaxEncodedPixelRate}")
+
+        self.step(8)
+        try:
+            asserts.assert_greater(len(aRankedStreamPriorities), 0, "RankedVideoStreamPrioritiesList is empty")
+            asserts.assert_greater(len(aRateDistortionTradeOffPoints), 0, "RateDistortionTradeOffPoints is empty")
+            videoStreamAllocateCmd = commands.VideoStreamAllocate(
+                streamUsage=aRankedStreamPriorities[0],
+                videoCodec=aRateDistortionTradeOffPoints[0].codec,
+                minFrameRate=aVideoSensorParams.maxFPS,
+                maxFrameRate=aVideoSensorParams.maxFPS,
+                minResolution=aRateDistortionTradeOffPoints[0].resolution,
+                maxResolution=aRateDistortionTradeOffPoints[0].resolution,
+                minBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                maxBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                minFragmentLen=4000,
+                maxFragmentLen=4000,
+            )
+            await self.send_single_cmd(endpoint=endpoint, cmd=videoStreamAllocateCmd)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+            pass
+
+        self.step(9)
+        aAllocatedVideoStreams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.AllocatedVideoStreams
+        )
+        logger.info(f"Rx'd AllocatedVideoStreams: {aAllocatedVideoStreams}")
+        asserts.assert_equal(len(aAllocatedVideoStreams), 1, "The number of allocated video streams in the list is not 1")
+
+        self.step(10)
+        try:
+            notSupportedStreamUsage = next(
+                (e for e in cluster.Enums.StreamUsageEnum if e not in aRankedStreamPriorities),
+                cluster.Enums.StreamUsageEnum.kUnknownEnumValue,
+            )
+            videoStreamAllocateCmd = commands.VideoStreamAllocate(
+                streamUsage=notSupportedStreamUsage,
+                videoCodec=aRateDistortionTradeOffPoints[0].codec,
+                maxFrameRate=aVideoSensorParams.maxFPS,
```

Comment:
minFrameRate is mandatory in the command, won't this fail in Python execution because it appears to be missing here?

Reply:
Thank you for your comments. We fixed it with new comment